frontend/public/components/pdb/PDBForm.tsx

```diff
@@ -0,0 +1,308 @@
+    ) {
+      setLabels(SelectorInput.arrayify(selector));
+      setPreviousValue(SelectorInput.arrayify(selector));
+      onFormValuesChange({ ...formValues, selector });
+    }
+  }, [existingData, formValues, onFormValuesChange, previousValue, selector]);
+
+  const handleNameChange = (value: string) => onFormValuesChange({ ...formValues, name: value });
+
+  const handleSelectorChange = (labelValues: string[]) => {
+    onFormValuesChange({ ...formValues, selector: SelectorInput.objectify(labelValues) });
+    setLabels([...labels, ...labelValues]);
+    setMatchingSelector(existingData);
+  };
+
+  const handleAvailabilityRequirementKeyChange = (value: string) => {
+    setRequirement(value);
+    setIsDisabled(false);
+    onFormValuesChange({
+      ...formValues,
+      requirement: value,
+      minAvailable: '',
+      maxUnavailable: '',
+    });
+  };
+  const handleAvailabilityRequirementValueChange = (value: string | number) => {
+    onFormValuesChange({ ...formValues, [requirement]: value });
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setInProgress(true);
+
+    const patch = [
+      {
+        op: 'replace',
+        path: '/spec/selector/matchLabels',
+        value: formValues.selector,
+      },
+      existingResource?.spec?.minAvailable
+        ? {
+            op: 'remove',
+            path: '/spec/minAvailable',
+          }
+        : {
+            op: 'remove',
+            path: '/spec/maxUnavailable',
+          },
+      formValues.minAvailable
+        ? {
+            op: 'replace',
+            path: '/spec/minAvailable',
+            value: convertToInt(formValues.minAvailable),
+          }
+        : {
+            op: 'replace',
+            path: '/spec/maxUnavailable',
+            value: convertToInt(formValues.maxUnavailable),
+          },
+    ];
+    const httpMethod: (
+      kind: K8sKind,
+      resource: PodDisruptionBudgetKind,
+      patch?: any,
+    ) => Promise<PodDisruptionBudgetKind> = existingResource ? k8sPatch : k8sCreate;
+
+    const response = existingResource
+      ? httpMethod(PodDisruptionBudgetModel, existingResource, patch)
+      : httpMethod(PodDisruptionBudgetModel, pdbToK8sResource(formValues));
+    return response
+      .then(() => {
+        setInProgress(false);
+        history.push(
+          resourcePathFromModel(PodDisruptionBudgetModel, formValues.name, formValues.namespace),
+        );
+      })
+      .catch((err) => {
+        setError(err.message);
+        setInProgress(false);
+      });
+  };
+
+  return (
+    <div className="co-m-pane__body co-m-pane__form">
+      <Form onSubmit={handleSubmit}>
+        <Grid hasGutter md={6}>
+          <GridItem span={12}>
+            <FormGroup label={t('public~Name')} isRequired fieldId="name">
+              <TextInput
+                isRequired
+                type="text"
+                name="name"
+                aria-labelledby={t('public~name')}
```

Comment:
`aria-labelledby` should reference an element ID, not the actual label. I think you can leave it out here since the `FormGroup` adds a `label` with the correct `for` attribute.